openshift-kube-apiserver/filters/deprecatedapirequest/v1helpers/helpers.go

```diff
@@ -0,0 +1,189 @@
+			return
+		}
+		i := timestamp.Hour()
+		for len(status.RequestsLast24h) <= i {
+			status.RequestsLast24h = append(status.RequestsLast24h, apiv1.RequestLog{})
+		}
+		status.RequestsLast24h[i] = incrementDeprecatedApiRequestCount(status.RequestsLast24h[i], node, timestamp, verb, username)
+	}
+}
+
+func incrementDeprecatedApiRequestCount(log apiv1.RequestLog, nodeName string, timestamp time.Time, verb string, username string) apiv1.RequestLog {
+	var n int
+	if n = indexOfNode(log.Nodes, nodeName); n < 0 {
+		log.Nodes = appendNodeRequestLog(log.Nodes, apiv1.NodeRequestLog{NodeName: nodeName})
+		n = indexOfNode(log.Nodes, nodeName)
+	}
+	var u int
+	if u = indexOfUser(log.Nodes[n].Users, username); u < 0 {
+		log.Nodes[n].Users = appendRequestUser(log.Nodes[n].Users, apiv1.RequestUser{UserName: username})
+		u = indexOfUser(log.Nodes[n].Users, username)
+	}
+	var c int
+	if c = indexOfCount(log.Nodes[n].Users[u].Requests, verb); c < 0 {
+		log.Nodes[n].Users[u].Requests = appendRequestCount(log.Nodes[n].Users[u].Requests, apiv1.RequestCount{Verb: verb})
+		c = indexOfCount(log.Nodes[n].Users[u].Requests, verb)
+	}
+	log.Nodes[n].Users[u].Requests[c].Count++
+	log.Nodes[n].Users[u].Count++
+	if log.Nodes[n].LastUpdate.Time.Before(timestamp) {
+		log.Nodes[n].LastUpdate = metav1.NewTime(timestamp)
+	}
+	return log
+}
+
+func appendRequestCount(s []apiv1.RequestCount, e ...apiv1.RequestCount) []apiv1.RequestCount {
+	s = append(s, e...)
+	sort.SliceStable(s, func(i, j int) bool {
+		return strings.Compare(s[i].Verb, s[j].Verb) < 0
+	})
+	return s
+}
+
+func appendRequestUser(s []apiv1.RequestUser, e ...apiv1.RequestUser) []apiv1.RequestUser {
+	s = append(s, e...)
+	sort.SliceStable(s, func(i, j int) bool {
+		return strings.Compare(s[i].UserName, s[j].UserName) < 0
+	})
+	return s
+}
+
+func appendNodeRequestLog(s []apiv1.NodeRequestLog, e ...apiv1.NodeRequestLog) []apiv1.NodeRequestLog {
+	s = append(s, e...)
+	sort.SliceStable(s, func(i, j int) bool {
+		return strings.Compare(s[i].NodeName, s[j].NodeName) < 0
+	})
+	return s
+}
+*/
+// RotateNodeRequestLogs rotates logs for a given node from status.RequestsLastHour to status.RequestsLast24h when needed.
+func RotateNodeRequestLogs(nodeName string, currentTime time.Time) UpdateStatusFunc {
+	return func(status *apiv1.DeprecatedAPIRequestStatus) {
+		currentHour := currentTime.Hour()
+		activeLogs := &status.RequestsLastHour.Nodes
+		activeLogIndex := indexOfNode(*activeLogs, nodeName)
+		if activeLogIndex < 0 {
+			klog.V(6).Infof("requestsLastHour does not contain request logs for node %s", nodeName)
+			return
+		}
+
+		// archive the requestsLastHour node log to the requestsLast24h node log
+		activeLog := status.RequestsLastHour.Nodes[activeLogIndex]
+		activeLogHour := activeLog.LastUpdate.Hour()
+		if activeLogHour != currentHour {
+
+			// ensure the hour index exists
+			for len(status.RequestsLast24h) <= activeLogHour {
```

Comment:
if the server was off for two hours, would adding a single item still work?  unit test?